src/main/java/core/basesyntax/service/DataValidation.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.TransactionDto;
+
+public interface DataValidation {
+    void validation(TransactionDto transactionDto);
```

Comment:
let's remove this interface and its impl class

and move the quantity check for purchase into PurchaseStrategy